app/policies/subject_rule_effect_policy.rb

```diff
@@ -38,26 +21,38 @@ def edit?
   end
 
   def update?
-    # record is a subject_rule_effect from the controller
-    if credential.admin?
-      true
-    else
-      # this is a good place to check the defined subect set id in the
-      # rule effect belongs to the scoped credential project ids using the API client
+    return true if credential.admin?
+    return false unless valid_credentials?
+
+    if record.config.key?('subject_set_id')
+      subject_set = Effects.panoptes.subject_set(record.config['subject_set_id'])
+      raise ActiveRecord::RecordNotFound if subject_set.nil?
```

Comment:
not blocking just another way to write it
```suggestion
      subject_set = Effects.panoptes.subject_set(record.config['subject_set_id'])
      raise ActiveRecord::RecordNotFound unless subject_set
```

Reply:
To be honest, I am partial to the way it's already written, just feels more clear/straightforward/explicit to me.